[DOC] Add html strip processor documentation (#5984)
opensearch-project/documentation-website: _ingest-pipelines/processors/html-strip.md

---
layout: default
title: HTML strip
parent: Ingest processors
nav_order: 140
---

# HTML strip processor

The `html_strip` processor removes HTML tags from string fields in incoming documents. The processor is useful when indexing data from web pages or other sources that may contain HTML markup. By removing the HTML tags, you can ensure that the indexed content is clean and easily searchable. HTML tags are replaced with newline characters (`\n`).

The following is the syntax for the `html_strip` processor:

```json

> <insert syntax example>

Reviewer: ```
{
  "html_strip": {
    "field": "webpage"
  }
}
```

`target_field` | Optional | The field to assign the cleaned value to. If not specified, field is updated in-place.
`ignore_missing` | Optional | Default is `false`. If `true`, the processor quietly exits without modifying the document when field does not exist.
`description` | Optional | Description of the processor's purpose or configuration.
`if` | Optional | Conditionally execute the processor.
`ignore_failure` | Optional | Ignore failures for the processor. See [Handling pipeline failures]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/).
`on_failure` | Optional | Handle failures for the processor. See [Handling pipeline failures]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/).
`tag` | Optional | Identifier for the processor. Useful for debugging and metrics.

## Using the processor

Follow these steps to use the processor in a pipeline.

### Step 1: Create a pipeline

The following query creates a pipeline named `strip-html-pipeline` that uses the `html_strip` processor to remove HTML tags from the description field and store the processed value in a new field named `cleaned_description`:

```json
PUT _ingest/pipeline/strip-html-pipeline
{

>   "description": "A pipeline to strip HTML from description field",

Reviewer: This example looks good to me.

---
layout: default
title: HTML strip
parent: Ingest processors
nav_order: 140
---

> # HTML strip processor

Reviewer 2: Global: Ensure H1 formatting consistency across these docs. gsub did not use code font, but script did, and we do not use it here.

Author: Revised across the latest ones to use regular font instead of code font. Published docs use reg font.

> The `html_strip` processor removes HTML tags from string fields in incoming documents. The processor is useful when indexing data from web pages or other sources that may contain HTML markup. By removing the HTML tags, you can ensure that the indexed content is clean and easily searchable. HTML tags are replaced with newline characters (`\n`).

Reviewer 2: ```suggestion
The `html_strip` processor removes HTML tags from string fields in incoming documents. This processor is useful when indexing data from webpages or other sources that may contain HTML markup. By removing the HTML tags, you can ensure that the indexed content is clean and easily searchable. HTML tags are replaced with newline characters (`\n`).
```

> The `html_strip` processor removes HTML tags from string fields in incoming documents. The processor is useful when indexing data from web pages or other sources that may contain HTML markup. By removing the HTML tags, you can ensure that the indexed content is clean and easily searchable. HTML tags are replaced with newline characters (`\n`).

Reviewer 2: Can we be more precise than "clean"? What do we actually mean by this?

> The `html_strip` processor removes HTML tags from string fields in incoming documents. This processor is useful when indexing data from webpages or other sources that may contain HTML markup. By removing the HTML tags, you can ensure that the indexed content is clean and easily searchable. HTML tags are replaced with newline characters (`\n`).

Author: ```suggestion
The `html_strip` processor removes HTML tags from string fields in incoming documents. This processor is useful when indexing data from webpages or other sources that may contain HTML markup. HTML tags are replaced with newline characters (`\n`).
```

The following is the syntax for the `html_strip` processor:

```json
{  
  "html_strip": {  
    "field": "webpage"  
  }  
}  
```
{% include copy-curl.html %}

## Configuration parameters

The following table lists the required and optional parameters for the `html_strip` processor.

Parameter | Required/Optional | Description |
|-----------|-----------|-----------|
`field` | Required | The string field from which to remove HTML tags.

> `target_field` | Optional | The field to assign the cleaned value to. If not specified, then the field is updated in-place.

Author: ```suggestion
`target_field` | Optional | The field to receive the plain text version after stripping HTML tags. If not specified, then the field is updated in-place.
```

Author: @natebower Please see revision made to clarify "cleaned value."

Reviewer 2: LGTM, but "to receive" => "that receives"?

> `target_field` | Optional | The field to assign the cleaned value to. If not specified, field is updated in-place.

Reviewer 2: ```suggestion
`target_field` | Optional | The field to assign the cleaned value to. If not specified, then the field is updated in-place.
```

{: .tip}

To test the pipeline, run the following query:

```json
POST _ingest/pipeline/strip-html-pipeline/_simulate
{
  "docs": [
    {
      "_source": {
        "description": "This is a <b>test</b> description with <i>some</i> HTML tags."
      }
    }
  ]
}
```
{% include copy-curl.html %}

#### Response

The following example response confirms that the pipeline is working as expected:

```json
{
  "docs": [
    {
      "doc": {
        "_index": "_index",
        "_id": "_id",
        "_source": {
          "description": "This is a <b>test</b> description with <i>some</i> HTML tags.",
          "cleaned_description": "This is a test description with some HTML tags."
        },
        "_ingest": {
          "timestamp": "2024-05-22T21:46:11.227974965Z"
        }
      }
    }
  ]
}
```
{% include copy-curl.html %}

### Step 3: Ingest a document 

The following query ingests a document into an index named `products`:

```json
PUT products/_doc/1?pipeline=strip-html-pipeline
{
  "name": "Product 1",
  "description": "This is a <b>test</b> product with <i>some</i> HTML tags."
}
```
{% include copy-curl.html %}

#### Response

> The response shows that the request has indexed the document into the index `products` and will index all documents with the `description` field containing HTML tags, while storing the cleaned version in the `cleaned_description` field.

Reviewer 2: ```suggestion
The response shows that the request has indexed the document into the index `products` and will index all documents with the `description` field containing HTML tags while storing the clean version in the `cleaned_description` field:
```

> The response shows that the request has indexed the document into the index `products` and will index all documents with the `description` field containing HTML tags, while storing the cleaned version in the `cleaned_description` field.

Reviewer 2: Same comment re: "clean"

```json
{
  "_index": "products",
  "_id": "1",
  "_version": 1,
  "result": "created",
  "_shards": {
    "total": 2,
    "successful": 1,
    "failed": 0
  },
  "_seq_no": 0,
  "_primary_term": 1
}
```
{% include copy-curl.html %}

### Step 4 (Optional): Retrieve the document

To retrieve the document, run the following query:

```json
GET products/_doc/1
```
{% include copy-curl.html %}

#### Response

> The response includes both the original `description` field and the `cleaned_description` field with HTML tags removed.

Reviewer 2: ```suggestion
The response includes both the original `description` field and the `cleaned_description` field with HTML tags removed:
```
